packages/model-viewer/src/features/annotation.ts

```diff
@@ -194,6 +201,45 @@ export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
       hotspot.updateNormal(config.normal);
     }
 
+    /**
+     * This method returns the world position and normal of the point on the
+     * mesh corresponding to the input pixel coordinates given relative to the
+     * model-viewer element. The position and normal are returned as strings in
+     * the format suitable for putting in a hotspot's data-position and
+     * data-normal attributes. If the mesh is not hit, position returns the
+     * empty string.
+     */
+    positionAndNormalFromPoint(pixelX: number, pixelY: number):
+        {position?: Vector3D, normal?: Vector3D} {
+      const {width, height} = this[$scene];
+      this[$pixelPosition]
+          .set(pixelX / width, pixelY / height)
+          .multiplyScalar(2)
+          .subScalar(1);
+      this[$pixelPosition].y *= -1;
+      raycaster.setFromCamera(this[$pixelPosition], this[$scene].getCamera());
+      const hits = raycaster.intersectObject(this[$scene], true);
+
+      if (hits.length === 0) {
+        return {};
+      }
+      const hit = hits[0];
+      const worldToPivot =
+          new Matrix4().getInverse(this[$scene].pivot.matrixWorld);
+      const position = new Vector3D();
+      Object.assign(position, hit.point.applyMatrix4(worldToPivot));
+
+      if (hit.face == null) {
```

Comment:
What is this condition?

Reply:
three can hit things that aren't faces. I think it would be rare for us, but I don't want to put an arbitrary return normal.